subsys/net/ip/net_if.c

```diff
@@ -190,11 +199,12 @@ static void net_if_process_events(struct k_poll_event *event, int ev_count)
 			break;
 		case K_POLL_STATE_FIFO_DATA_AVAILABLE:
 		{
-			struct net_if *iface;
+			struct net_if_dev *if_dev;
+
+			if_dev = CONTAINER_OF(event->fifo, struct net_if_dev,
+					      tx_queue);
 
-			iface = CONTAINER_OF(event->fifo, struct net_if,
-					     tx_queue);
-			net_if_tx(iface);
+			net_if_foreach(iface_tx_cb, if_dev);
```

Comment:
Ok, this is temporary as new PR will change that entirely (TC support etc...) So it won't need to loop on each iface etc...